internal/web/routers/purchase-router.go

```diff
@@ -7,4 +7,5 @@ import (
 
 func SetupPurchasesRouter(r chi.Router, controller *controllers.PurchaseController) {
 	r.Post("/purchases/finish", controller.FinishPurchase)
+	r.Get("/purchases/{user_id}", controller.GetPurchasesByUser)
```

Comment:
A sintaxe dessa rota está mais adequada para uma rota de detalhes de compra.
Talvez algo assim deixe um sentido melhor:
```suggestion
	r.Get("users/{user_id}/purchases", controller.GetPurchasesByUser)
```
Apesar de ser o roteador que cuida das rotas referentes a compras, não necessariamente a rota precisa começar com `/purchases`, o mais importante de uma rota é ela comunicar para que ela serve.